### PR TITLE
feat: support new AWS regions [DET-3837]

### PR DESCRIPTION
**Pull request by rb-determined-ai**
## Description

This is the proposed change to follow-up https://github.com/determined-ai/environments/pull/49.

It includes an application of `bumpenvs.py` that I apparently lost during a rebase in the PR where I added `bumpenvs.py`.

### CODE REVIEW

**Comment by rb-determined-ai** [comment]
@seanr15 please review the 4th commit specifically (the others are mostly just automated scripts)

**Comment by seanr15** [comment]
Looks good so far. You might want to check to make sure the added regions support p2 and p3 instances.

**Comment by rb-determined-ai** [comment]
> You might want to check to make sure the added regions support p2 and p3 instances.

Check!  I added the P3 regions (minus China, minus US gov regions), and the only one which didn't have `p2.8xlarge` instances was London.

**Comment by rb-determined-ai** [comment]
@aaron276h you are up next.  I wanted you to have a look at the multi-region support in this repo before I land https://github.com/determined-ai/environments/pull/49

**Comment by aaron276h** [comment]
@rb-determined-ai everything here looks g2m

**Comment by rb-determined-ai** [comment]
@aaron276h the diff here from what you saw before is that there is a new script called `refresh-ubuntu-amis.py`, since I found out that we had literally no mechanism in place for updating the default ubuntu AMIs that we used for master and bastion.

File: tools/scripts/bumpenvs.yaml [hunk 1]
```diff
@@ -1,33 +1,83 @@
...
-  old: determinedai/environments:py-3.6.9-pytorch-1.4-tf-2.2-cpu-aaa3750
+  new: determinedai/environments:py-3.6.9-pytorch-1.4-tf-2.2-cpu-7b8b1e9
+  old: determinedai/environments:py-3.6.9-pytorch-1.4-tf-2.2-cpu-9b346b2
 tf2_cpu_versioned:
   new: determinedai/environments:py-3.6.9-pytorch-1.4-tf-2.2-cpu-0.5.0
 tf2_gpu_hashed:
-  new: determinedai/environments:cuda-10.1-pytorch-1.4-tf-2.2-gpu-06318c7
-  old: determinedai/environments:cuda-10.1-pytorch-1.4-tf-2.2-gpu-aaa3750
+  new: determinedai/environments:cuda-10.1-pytorch-1.4-tf-2.2-gpu-7b8b1e9
+  old: determinedai/environments:cuda-10.1-pytorch-1.4-tf-2.2-gpu-9b346b2
```
Comment (aaron276h) [inline tools/scripts/bumpenvs.yaml:63]: where is the `9b346b2` coming from?

File: tools/scripts/README.md [hunk 2]
```diff
@@ -30,6 +30,11 @@ be registered in the determined repo. Here is the process:
    artifact's `old` value to the previous `new` value, and set the new `new`
    value to the artifact produced by CI.
 
-5. Run `./bumpenvs.py bumpenvs.yaml`.  This will do a simple string replacement
+5. (optional) Run `./refresh-ubuntu-amis.py bumpenvs.yaml`.  This will fetch
```
Comment (aaron276h) [inline tools/scripts/README.md:33]: just to make sure, agent's ami are handled as part of step 4?

File: tools/scripts/bumpenvs.yaml (hunk 1 above)
Comment (rb-determined-ai) [reply tools/scripts/bumpenvs.yaml:63]: There are 3 different applications of bumpenvs in this commit.  You can see them in the individual commits.  There was one I missed from before, followed by the remove-PEDL update, followed by the multi-region update.

The reason I split up the last two was to purge the `pedl-` tags from the bumpenvs file.

Comment (rb-determined-ai) [reply tools/scripts/bumpenvs.yaml:63]: Side note: there's no meaningful difference in the actual contents of any of these changes, so if we cut a release without the most up-to-date AMIs it's not too big of a deal.

File: tools/scripts/README.md (hunk 2 above)
Comment (rb-determined-ai) [reply tools/scripts/README.md:33]: Yeah, the agent amis are custom things built by the `environments repo`, but the master and bastion amis are totally default ubuntu images, and AFAICT have never been updated since we first wrote these templates.

File: tools/scripts/README.md [hunk 3]
```diff
@@ -30,6 +30,11 @@ be registered in the determined repo. Here is the process:
    artifact's `old` value to the previous `new` value, and set the new `new`
    value to the artifact produced by CI.
```
Comment (aaron276h) [inline tools/scripts/README.md:31]: non-blocking: not part of this PR, but could help to call that this typically includes task containers and agent AMIs.

**Review by aaron276h — APPROVED** [review]
🚢 🚢 🚢

File: tools/scripts/refresh-ubuntu-amis.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+import yaml
+
+get_cache = {}  # type: Dict[str, bytes]
+
+
+def cacheable_get(url: str) -> bytes:
+    global get_cache
+    if url not in get_cache:
+        req = requests.get(url)
+        assert req.ok, f"request to {url} failed with HTTP {req.status_code}"
```
Comment (dzhu) [inline tools/scripts/refresh-ubuntu-amis.py:24]: non-blocking: `req.raise_for_status()`, perhaps.

File: tools/scripts/refresh-ubuntu-amis.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        get_cache[url] = req.content
```
Comment (dzhu) [inline tools/scripts/refresh-ubuntu-amis.py:25]: non-blocking: Use `req.text` to skip the `decode()` later.

File: tools/scripts/refresh-ubuntu-amis.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    return get_cache[url]
+
+
+def get_ubuntu_ami(release: str, region: str) -> str:
+    resp = cacheable_get(
+        f"https://cloud-images.ubuntu.com/query/{release}/server/released.current.txt"
+    ).decode("utf8")
+
+    ami_lines = [line.split("\t") for line in resp.rstrip("\n").split("\n")]
```
Comment (dzhu) [inline tools/scripts/refresh-ubuntu-amis.py:34]: non-blocking: `resp.splitlines()` instead (unless you specifically want to account for the possibility of multiple trailing newlines).

non-blocking: Use of `namedtuple` would be pleasing (but not very necessary due to the small scope here).

File: tools/scripts/refresh-ubuntu-amis.py [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+    if len(sys.argv) < 2:
+        print(__doc__, file=sys.stderr)
+        sys.exit(1)
+
+    path = sys.argv[1]
+
+    with open(path) as f:
+        conf = yaml.safe_load(f)
+
+    for image_type in conf:
```
Comment (dzhu) [inline tools/scripts/refresh-ubuntu-amis.py:76]: non-blocking: Could be `for image_type, subconf in conf.items()` to save the subscripting later.

**Review by dzhu — APPROVED** [review]

File: tools/scripts/refresh-ubuntu-amis.py (hunk 4 above)
Comment (rb-determined-ai) [reply tools/scripts/refresh-ubuntu-amis.py:24]: Ah, I looked for that function but couldn't find it and assumed I was think of some other library : )

File: tools/scripts/README.md (hunk 3 above)
Comment (rb-determined-ai) [reply tools/scripts/README.md:31]: done

File: tools/scripts/refresh-ubuntu-amis.py (hunk 5 above)
Comment (rb-determined-ai) [reply tools/scripts/refresh-ubuntu-amis.py:25]: done

File: tools/scripts/refresh-ubuntu-amis.py (hunk 4 above)
Comment (rb-determined-ai) [reply tools/scripts/refresh-ubuntu-amis.py:24]: done

File: tools/scripts/refresh-ubuntu-amis.py (hunk 6 above)
Comment (rb-determined-ai) [reply tools/scripts/refresh-ubuntu-amis.py:34]: re: splitlines: there are too many builtin methods in python.  But I'm glad to learn about this one.

re namedtuple: There's too many fields that I don't know how to name... I thought this came out nicer (I tried namedtuple first actually)

File: tools/scripts/refresh-ubuntu-amis.py (hunk 7 above)
Comment (rb-determined-ai) [reply tools/scripts/refresh-ubuntu-amis.py:76]: done.
